azurerm/resource_arm_servicebus_topic_authorization_rule.go

```diff
@@ -61,27 +69,41 @@ func resourceArmServiceBusTopicAuthorizationRuleCreateUpdate(d *schema.ResourceD
 	topicName := d.Get("topic_name").(string)
 	resourceGroup := d.Get("resource_group_name").(string)
 
+	if d.IsNewResource() {
```

Comment:
What's the motivation to add CheckIfExist logic on client code instead of leveraging server side logic? Better error handling logic here ?

If it's the new resource, it will call GET API one more time.

Reply:
> What's the motivation to add CheckIfExist logic on client code instead of leveraging server side logic? Better error handling logic here ?

Because the majority of the Azure API's are Upserts, doing a Create rather than an Upsert means weird behaviours are possible. Other cloud provider API's fail when you try and create a resource already exists - but unfortunately this behaviour isn't present in Azure, such that we've had multiple issues with users having weird bugs because they're not maintaining the Statefile. This PR brings the Azure Provider into line with the other Providers by requiring existing resources are imported (which means things like the `ForceNew` field can work as expected)

> If it's the new resource, it will call GET API one more time.

This is unfortunate, but I don't see a way around this given the Azure API's are Upsertable, unfortunately.